cybersecuritytools/csls/generate_cloudwatch_logs/put_cloudwatch_logs.py

```diff
@@ -0,0 +1,157 @@
+"""Module for generating test data of various types, send data to Cloudwatch."""
+import json
+from dataclasses import dataclass
+from datetime import datetime, timezone
+from functools import lru_cache
+from typing import Any, Dict, List
+from uuid import uuid4
+
+import boto3
+import mypy_boto3_logs as logs
+from mypy_boto3_logs.type_defs import PutLogEventsResponseTypeDef
+
+
+class LogEventResponseReal(PutLogEventsResponseTypeDef):
+    ResponseMetadata: Dict[Any, Any]
+
+
+@lru_cache(maxsize=None)
+def logs_client() -> logs.CloudWatchLogsClient:
+    return boto3.client("logs")
+
+
+@dataclass
+class LogLines:
+    logs: Dict[str, str]
+    payload: str
+    datetime: datetime
+
+
+def log_lines(payload: str = "") -> LogLines:
+    """The log lines to send to CloudWatch logs. This generates all the
+    supported log formats with an embedded UUID which can be used to
+    identify the logs at a later date.
+
+    Parameters
+    ----------
+    payload: A custom payload to add into the generated log lines
+
+    """
+    payload = payload or str(uuid4())
+    now = datetime.now()
+
+    logs = {
+        "raw": f"Hello {payload} {now}",
+        "json": json.dumps({"Hello": payload, "time": str(now)}),
+        "csv": (f"An Example [{payload}] logged in,192.168.1.1," f'"{now}",Unknown'),
+        "syslog": (
+            f"{now.strftime('%b %d %H:%M:%S')} ip-192-168-1-1 bash[1644]: "
+            f"#01 [{payload}] java.lang.Thread.run(Thread.java:748) [?:1.8.0_172]"
+        ),
+    }
+    return LogLines(logs, payload, now)
+
+
+@lru_cache()
+def log_formats() -> List[str]:
+    """Logging output formats"""
+    return [*log_lines().logs]
```

Comment:
same as above, isn't this just a top-level constant?

Reply:
As above, I've removed the `lru_cache`.